### PR TITLE
Port doctor CLI tests to conda_cli fixture.

### PR DESCRIPTION
**Pull request by jezdez**
### Description

Just some minor fixes I found while working on https://github.com/conda/conda/pull/12911, which removes the PendingDeprecationWarning in the test output

### Checklist - did you ...

- ~[ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: tests/plugins/subcommands/doctor/test_cli.py [hunk 1]
```diff
@@ -1,42 +1,46 @@
...
 
+from conda.testing import CondaCLIFixture, TmpEnvFixture
 
-def test_conda_doctor_happy_path():
+
+def test_conda_doctor_happy_path(conda_cli: CondaCLIFixture):
     """Make sure that we are able to call the ``conda doctor`` command"""
 
-    out, err, code = run(f"conda doctor")
+    out, err, code = conda_cli(f"doctor")
```
Comment (kenodegard) [inline tests/plugins/subcommands/doctor/test_cli.py:11]:
```suggestion
    out, err, code = conda_cli("doctor")
```

File: tests/plugins/subcommands/doctor/test_cli.py [hunk 2]
```diff
@@ -1,42 +1,46 @@
...
     assert not err  # no error message
     assert not code  # successful exit code
 
 
-def test_conda_doctor_happy_path_verbose():
+def test_conda_doctor_happy_path_verbose(conda_cli: CondaCLIFixture):
     """Make sure that we are able to run ``conda doctor`` command with the --verbose flag"""
 
-    out, err, code = run(f"conda doctor --verbose")
+    out, err, code = conda_cli(f"doctor", "--verbose")
```
Comment (kenodegard) [inline tests/plugins/subcommands/doctor/test_cli.py:20]:
```suggestion
    out, err, code = conda_cli("doctor", "--verbose")
```

File: tests/plugins/subcommands/doctor/test_cli.py [hunk 3]
```diff
@@ -1,42 +1,46 @@
...
     """Make sure that we are able to run ``conda doctor`` command with the --help flag"""
+    with pytest.raises(SystemExit):
+        out, err, code = conda_cli("doctor", "--help")
 
-    out, err, code = run(f"conda doctor --help")
-
-    assert "Display a health report for your environment." in out
-    assert not err  # no error message
-    assert not code  # successful exit code
+        assert "Display a health report for your environment." in out
```
[the remark this answers is not in the document]
Comment (jezdez) [reply tests/plugins/subcommands/doctor/test_cli.py:31]: Trying it out locally, the `match` parameter doesn't work as it just is able to test against the return of of `0` from `SystemExit` 🤷

**Review by ForgottenProgramme — DISMISSED** [review]

File: tests/plugins/subcommands/doctor/test_cli.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jezdez) [reply tests/plugins/subcommands/doctor/test_cli.py:31]: True!

**Review by kenodegard — APPROVED** [review]

**Review by beeankha — APPROVED** [review]

**Pull request merged**